### PR TITLE
feat(editor): add Markdown rendering with inline alignment

### PR DESCRIPTION
**Pull request by nebrask**
Created a Markdown rendering editor with the following features:

- **Inline Markdown Rendering**:
  - Adds an inline Markdown extension using **CodeMirror** for headings, bold, italic, lists, and block quotes.
  - Renders Markdown content with **markdown-it**, ensuring alignment with the associated line numbers in the editor.

- **Styling Adjustments**:
  - Aligns headings, lists, and block quotes in line with line numbers.
  - Removes unnecessary spacing above and below-rendered content.

### Relevant Changes
1. **EditorWithToolbar.tsx**:
   - Integrates the `inlineMarkdownExtension` for rendering Markdown.
2. **MarkdownRenderer.tsx**:
   - Implements the custom Markdown rendering logic using **CodeMirror** and **markdown-it**.
3. **index.css**:
   - Adds styling to align Markdown elements (headings, lists, block quotes, and plain text) properly.

### CODE REVIEW

File: packages/morph/package.json [hunk 1]
```diff
@@ -10,16 +10,22 @@
...
   },
   "dependencies": {
+    "@codemirror/lang-markdown": "^6.3.1",
+    "@codemirror/state": "^6.5.0",
+    "@codemirror/view": "^6.36.1",
     "class-variance-authority": "^0.7.1",
     "clsx": "^2.1.1",
+    "codemirror": "^6.0.1",
     "lucide-react": "^0.469.0",
+    "markdown-it": "^14.1.0",
```
Comment (aarnphm) [inline packages/morph/package.json:20]: markdown-it's plugins ecosystem is not that nice to write imo.

if we want to expand upon the core commonmark then I don't think we should use it.

File: packages/morph/src/index.css [hunk 2]
```diff
@@ -1,137 +1,71 @@
...
-
-  color-scheme: light dark;
-  color: rgba(255, 255, 255, 0.87);
-  background-color: #242424;
-
-  font-synthesis: none;
-  text-rendering: optimizeLegibility;
-  -webkit-font-smoothing: antialiased;
-  -moz-osx-font-smoothing: grayscale;
+.cm-editor .cm-scroller .cm-content .cm-line {
```
Comment (aarnphm) [inline packages/morph/src/index.css:1]: quick ask, can we use sass instead?

**Review by aarnphm — APPROVED** [review]
few comments

File: packages/morph/src/index.css (hunk 2 above)
Comment (nebrask) [reply packages/morph/src/index.css:1]: Never used it before but I'll look into it.

**Pull request closed without merging**